FirebaseStorage/Sources/FIRStorageDeleteTask.m

```diff
@@ -72,10 +70,11 @@ - (void)enqueue {
       }
       self->_fetcherCompletion = nil;
     };
-#pragma clang diangostic pop
 
     [fetcher beginFetchWithCompletionHandler:^(NSData *_Nullable data, NSError *_Nullable error) {
-      weakSelf.fetcherCompletion(data, error);
+      if (weakSelf.fetcherCompletion != nil) {
```

Comment:
This is unsafe--the object can be deallocated while the method is in progress. Calling a method does not retain self. You need to assign from weakSelf to a (new) strong reference and then check if that strong reference is nil.